### PR TITLE
udc: it82xx2: resolves several issues

### PR DESCRIPTION
**Pull request by Chenhongren**
This PR addresses several issues, including:
- fix the fifo control issue
- fix issue with the unexpected setting of the ready bit
- correct the handling of the out endpoint
- clear the data seq bit when enabling non-ctrl IN endpint
- enable RESUME interrupt
- emit SOF event

Tested by:
- samples/subsys/usb/console/
- samples/subsys/usb/cdc_acm/
- Chapter9 test
- Google Spikyrock project

### CODE REVIEW

**Review by tmon-nordic — DISMISSED** [review]

File: drivers/usb/udc/udc_it82xx2.c [hunk 1]
```diff
@@ -1119,6 +1118,23 @@ static inline int work_handler_out(const struct device *dev, uint8_t ep)
...
 
+	LOG_DBG("Handle data OUT, %zu | %zu", len, net_buf_tailroom(buf));
+
+	if (net_buf_tailroom(buf) && len == udc_mps_ep_size(ep_cfg)) {
+		work_handler_xfer_continue(dev, ep, buf);
+		if (ep != USB_CONTROL_EP_OUT) {
+			err = udc_submit_ep_event(dev, buf, 0);
+		}
+		return err;
+	}
```
Comment (jfischer-no) [inline drivers/usb/udc/udc_it82xx2.c:1151]: That looks suspicious. Does it start a new transaction from the thread context?

File: drivers/usb/udc/udc_it82xx2.c [hunk 2]
```diff
@@ -1394,7 +1404,11 @@ static void suspended_handler(struct k_work *item)
 	}
 
 	it82xx2_enable_resume_int(dev, true);
+
+#if (!defined(USB_DEVICE_CONFIG_SOF_NOTIFICATIONS) || (USB_DEVICE_CONFIG_SOF_NOTIFICATIONS == 0U))
```
Comment (jfischer-no) [inline drivers/usb/udc/udc_it82xx2.c:1408]: There should not be any `#if (!defined(USB_DEVICE_CONFIG_SOF_NOTIFICATIONS) || (USB_DEVICE_CONFIG_SOF_NOTIFICATIONS == 0U))`
```c
	if (USB_DEVICE_CONFIG_SOF_NOTIFICATIONS) {
		it82xx2_enable_sof_int(dev, true);
	}
```

**Review by jfischer-no — CHANGES_REQUESTED** [review]

File: drivers/usb/udc/udc_it82xx2.c (hunk 2 above)
Comment (Chenhongren) [reply drivers/usb/udc/udc_it82xx2.c]: Updated.

File: drivers/usb/udc/udc_it82xx2.c (hunk 1 above)
Comment (Chenhongren) [reply drivers/usb/udc/udc_it82xx2.c:1149]: Yes, is there any risk in starting a new OUT transaction within thread? Perhaps there's something I'm not fully understanding.

Comment (jfischer-no) [reply drivers/usb/udc/udc_it82xx2.c:1149]: There is actually no need, it slows the transfer. The usual approach is to start the first transaction from the thread context, handle any subsequent transactions in the ISR context, and notify the thread when finished.

**Review by jfischer-no — APPROVED** [review]

File: drivers/usb/udc/udc_it82xx2.c (hunk 1 above)
Comment (Chenhongren) [reply drivers/usb/udc/udc_it82xx2.c:1149]: Thanks for the feedback. This is now on our TODO list, and we will submit a PR if we make any improvements.

**Comment by Chenhongren** [comment]
@tmon-nordic Could you please approve the changes again if they look good to you? thanks.

**Review by Dino-Li — APPROVED** [review]

**Pull request merged**